stdlib/src/builtin/value.mojo

```diff
@@ -179,3 +179,37 @@ trait RepresentableCollectionElement(CollectionElement, Representable):
     """
 
     pass
+
+
+trait Indexer:
+    """This trait denotes a type that can be used to index a container that
+    handles integral index values.
+
+    This solves the issue of being able to index data structures such as `List` with the various
+    integral types without being too broad and allowing types that should not be used such as float point
+    values.
+    """
+
+    fn __index__(self) -> Int:
+        """Return the index value
+
+        Returns:
+            The index value of the object
+        """
```

Comment:
@JoeLoser Do we have styleguide recommendations on documentating method declarations in traits? And do they even generate docs correctly?